corehq/apps/userreports/tests/test_pillow.py

```diff
@@ -115,3 +101,38 @@ def _check_sample_doc_state(self, datetime_mock):
                 self.assertAlmostEqual(expected_indicators[k], v)
             else:
                 self.assertEqual(expected_indicators[k], v)
+
+
+class IndicatorConfigFilterTest(SimpleTestCase):
+
+    def setUp(self):
+        self.config = get_sample_data_source()
+
+    def test_filter(self):
+        not_matching = [
+            dict(doc_type="NotCommCareCase", domain='user-reports', type='ticket'),
+            dict(doc_type="CommCareCase", domain='not-user-reports', type='ticket'),
+            dict(doc_type="CommCareCase", domain='user-reports', type='not-ticket'),
+        ]
+        for document in not_matching:
+            self.assertFalse(self.config.filter(document)), 'Failing dog: %s' % document
+
+        self.assertTrue(self.config.filter(
+            dict(doc_type="CommCareCase", domain='user-reports', type='ticket')
+        ))
+
+    def test_deleted_filter(self):
+        not_matching = [
+            dict(doc_type="CommCareCase", domain='user-reports', type='ticket'),
+            dict(doc_type="CommCareCase-Deleted", domain='not-user-reports', type='ticket'),
+        ]
+        for document in not_matching:
+            self.assertFalse(self.config.deleted_filter(document), 'Failing dog: %s' % document)
+
+        matching = [
+            dict(doc_type="CommCareCase-Deleted", domain='user-reports', type='ticket'),
+            dict(doc_type="CommCareCase-Deleted", domain='user-reports', type='bot-ticket'),
+            dict(doc_type="CommCareCase-Deleted", domain='user-reports'),
```

Comment:
Matches docs that have the wrong type or no type but I don't think this is an issue since if the doc isn't in the table in the first place then the delete will be a no-op.

Could be a performance concern though but I don't think we have enough deletions for that to be the case.